app/src/frontend/building/data-container.tsx

```diff
@@ -73,45 +116,34 @@ const withCopyEdit = (WrappedComponent) => {
             })
         }
 
-        updateBuildingState(key, value) {
-            const building = {...this.state.building};
-            building[key] = value;
+        isEdited() {
+            const edits = this.state.buildingEdits;
+            // check if the edits object has any fields
+            return Object.entries(edits).length !== 0;
+        }
 
+        clearEdits() {
             this.setState({
-                building: building
+                buildingEdits: {}
             });
         }
 
-        /**
-         * Handle changes on typical inputs
-         * - e.g. input[type=text], radio, select, textare
-         *
-         * @param {*} event
-         */
-        handleChange(event) {
```

Comment:
no need for event-processing because input components provide the already parsed value for field (also deleted handleCheck for same reason)